### PR TITLE
User and Admin log in and sign up page

### PR DESCRIPTION
**Pull request by mumble-07**
### **THE STORY**

> This PR is for Trader: user story 1 and user story 2, and some RSPEC model and request testing. 
> This PR is for installing Devise gem for TRADER and ADMIN. Traders can create an account and log in their credentials. While Admins can only log in their credentials.

> 
_**NOTE:** Creating an Admin account will only be allowed via the console._

**Controllers**
 > A static_pages_controller -> home_page was generated to be the landing page of the app, and for other static pages to be used in the succeeding development of the project. 
> An admins_controller and users_controller was also generated.

**Models**
>A user and admin model was generated.

**Views**
> inside the devise generated views for admin and user, an index.html.erb was added for both user and admin to be the user/admin home page.

_ADDITIONAL NOTE: Mailer is not yet added in this PR._

### **REFERENCES**
Devise Gem: https://github.com/heartcombo/devise/wiki/How-to-Setup-Multiple-Devise-User-Models

### **SCREENSHOTS**

1. root page 
image
**
TRADER/USER

2. users page sign-up/log-in, when you click the I'm a trader link 
image

3. sign up
image

4. log in
image

ADMIN LOG IN ONLY

5.  Admin page log in when you click the I'm an Admin link
image

6. log in 
image

### CODE REVIEW

File: app/views/layouts/_alerts.html.erb [hunk 1]
```diff
@@ -0,0 +1,2 @@
+  <p class="notice"><%= notice %></p>
+  <p class="alert"><%= alert %></p>
```
Comment (analyn-cajocson) [inline app/views/layouts/_alerts.html.erb:2]: add new line at the end of file

File: app/views/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,8 @@
+<% if current_user %>
+	<h2> I'm a Trader </h2>
+	<%= link_to "sign out", destroy_user_session_path, method: :delete %>
+<% else %>
+	<h2> I'm a Trader </h2>
+	<%= link_to "Register", new_user_registration_path %>
+	<%= link_to "sign in", new_user_session_path %>
+<% end %>
```
Comment (analyn-cajocson) [inline app/views/users/index.html.erb:8]: Fix code alignment: 2 spaces
add new line at the end of file

File: db/migrate/20210911041706_add_wallet_id_to_admins.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+class AddWalletIdToAdmins < ActiveRecord::Migration[6.0]
+  def change
+    add_column :admins, :wallet_id, :integer
```
Comment (analyn-cajocson) [inline db/migrate/20210911041706_add_wallet_id_to_admins.rb:3]: Why do we nedd to add `wallet_id` to the `admins` table?

**Review by analyn-cajocson — CHANGES_REQUESTED** [review]

File: db/migrate/20210911041706_add_wallet_id_to_admins.rb (hunk 3 above)
Comment (mumble-07) [reply db/migrate/20210911041706_add_wallet_id_to_admins.rb:3]: We deleted the migration, got confused with our ERD. `wallet_id` is not necessary. Thanks for pointing that out. :)

**Review by michael-dollosa — COMMENTED** [review]
lgtm

**Review by analyn-cajocson — APPROVED** [review]

**Pull request merged**